### ISSUE TITLE
the 'examples' values should be honored when using type integer+oneOf

### ISSUE THREAD

**Issue by nospam2000** [asyncapi/html-template#195]
#### Describe the bug
The automatically created examples don't honor the specified values of 'examples' and create invalid combinations of attributes. The result is a logically invalid example.

#### How to Reproduce
definition:
```
    arrayValueRank:
      description: >
        This Attribute indicates whether the val Attribute of the datapoint is an
        array and how many dimensions the array has.
      type: integer
      default: -1
      examples:
        - 2
      oneOf:
        - const: -1
          description: 'Scalar: The value is not an array.'
        - const: 0
          description: 'OneOrMoreDimensions: The value is an array with one or more dimensions.'
        - const: 1
          description: 'OneDimension: The value is an array with one dimension.'
        - const: 2
          description: 'The value is an array with two dimensions.'

    arrayArrayDimensions:
      type: array
      items:
        type: integer
        minimum: 0
      examples:
        - [3, 5]
```

The created invalid example
```
              "valueRank": -1,
              "arrayDimensions": [
                3,
                5
              ],
```

'-1' means a scalar value, but for the example of arrayDimensions valueRank has to be '2'

#### Expected behavior
expected example:
```
              "valueRank": 2,
              "arrayDimensions": [
                3,
                5
              ],
```
use the value '2' as the only value provided in 'examples' for valueRank. Instead the first value of the valueOf is used.

Changing the order of the values in my schema will also confuse the user, because they shall be numerically sorted and the first value makes no sense as example.

**Comment by magicmatatjahu** [comment]
@nospam2000 Hello! Sorry for that late response. Problem is that we use the `openapi-sampler` package to generate examples from schema. As you can see https://github.com/Redocly/openapi-sampler/blob/master/src/traverse.js#L92 package infers examples in order from: `default` -> `const` -> `enum` -> `examples` field. We can only "fix" your problem by contribution to package.

**Comment by magicmatatjahu** [comment]
I created issue for that in mentioned package https://github.com/Redocly/openapi-sampler/issues/124

**Comment by magicmatatjahu** [comment]
I created PR to fix that bug https://github.com/Redocly/openapi-sampler/pull/126

**Comment by magicmatatjahu** [comment]
I used the latest `openapi-sampler` version and for provided schema it creates example:

![image](https://user-images.githubusercontent.com/20404945/124267597-c08c3e00-db38-11eb-9621-2c36d20bdaf0.png)

**Comment by magicmatatjahu** [comment]
Fixed in `v0.23.1` :)
